functions/index.js

```diff
@@ -34,28 +43,62 @@ exports.createNewTask = functions.https.onRequest((request, response) => {
         console.log(status);
         console.log(category);
         console.log(createNewTaskSprintNumber);
+        console.log(storyPointNumber);
 
-        db.collection(fullSprintId).doc(category).set({
-                Title: title,
-                Description: des,
-                Priority: priority,
-                Difficulty: difficulty,
-                Creator: creator,
-                Assignee: assignee,
-                ET: estimatedTime,
-                Status: status,
-                Category: category,
-                LogWorkTotalTime: loggedWorkTotalTime,
-                WorkDone: workDone,
-                CreateNewTaskSprintNumber: createNewTaskSprintNumber
+        db.collection("Main").doc("RawData").get().then((doc) => {
+
+                totalDevelopmentTask = doc.data().TotalDevelopmentTask + 1;
+                totalBusinessTask = doc.data().TotalBusinessTask + 1;
+                totalMarketingTask = doc.data().TotalBusinessTask + 1;
+                totalNumberOfTask = doc.data().TotalNumberOfTask;
+
+                if (category === "Development") {
+                    taskId = category[0] + totalDevelopmentTask;
+                } else if (category === "Business") {
+                    taskId = category[0] + totalBusinessTask;
+                } else {
+                    taskId = category[0] + totalMarketingTask;
+                }
+                console.log(taskId);
+
+                var setDataPromise = db.collection(fullSprintId).doc(taskId).set({
+                    Title: title,
+                    Description: des,
+                    Priority: priority,
+                    Difficulty: difficulty,
+                    Creator: creator,
+                    Assignee: assignee,
+                    ET: estimatedTime,
+                    Status: status,
+                    Category: category,
+                    LogWorkTotalTime: loggedWorkTotalTime,
+                    WorkDone: workDone,
+                    CreateNewTaskSprintNumber: createNewTaskSprintNumber,
+                    StoryPointNumber: storyPointNumber
+                });
+                return Promise.resolve(setDataPromise);
             })
-            .then(() => {
-                var work = { data: "working" }
+            .then(function(setDataPromise) {
+                var updateSetDataPromise = db.collection("Main").doc("RawData").update({
+                    TotalDevelopmentTask: totalDevelopmentTask,
+                    TotalBusinessTask: totalBusinessTask,
+                    TotalMarketingTask: totalMarketingTask,
+                    TotalNumberOfTask: totalNumberOfTask,
+                    TotalCompletedTask: totalCompletedTask,
+                    TotalUnCompletedTask: totalUnCompletedTask
+                });
+                return Promise.resolve(updateSetDataPromise);
+            })
```

Comment:
just like this one more place needs to be updated.

db --> "Main" --> fullSprintId 

And this location in db will not be updated in the same data.
Rather at first you need to get the data from here and use the logic to store data as in RawData

Reply:
Difference between RawData and fullSprintId:

RawData --> Contains the overall data of the application.
fullSprintId --> Contains the same data but only for the sprint number.

Reply:
no here fullSprint Id is the collection name

Reply:
And im not getting how to count number of completed and uncompleted task

Reply:
first here i have to count then only I can update